server/broadcast.go

```diff
@@ -346,12 +356,28 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 
 	// send segment to the orchestrator
 	glog.V(common.DEBUG).Infof("Submitting segment nonce=%d manifestID=%s seqNo=%d orch=%s", nonce, cxn.mid, seg.SeqNo, sess.OrchestratorInfo.Transcoder)
+	if sess.Sender != nil {
+		err := sess.Sender.ValidateTicketParams(pmTicketParams(sess.OrchestratorInfo.TicketParams))
+		if err != nil {
+			if err != pm.ErrTicketParamsExpired {
+				return nil, err
+			}
 
+			glog.V(common.VERBOSE).Infof("ticketparams expired, refreshing for orchestrator=%v", sess.OrchestratorInfo.Transcoder)
+			newSess, err := refreshSession(sess)
+			if err != nil {
+				cxn.sessManager.removeSession(sess)
+				return nil, fmt.Errorf("unable to refresh ticketparams for orchestrator=%v, err=%v", sess.OrchestratorInfo.Transcoder, err)
+			}
+			sess = newSess
+		}
+	}
 	res, err := SubmitSegment(sess, seg, nonce)
 	if err != nil || res == nil {
+		glog.Infof("unable to submit segment, removing session for orchestrator=%v err=%v", sess.OrchestratorInfo.Transcoder, err)
 		cxn.sessManager.removeSession(sess)
 		if res == nil && err == nil {
-			return nil, errors.New("Empty response")
+			return nil, fmt.Errorf("unable to submit segment, removing session for orchestrator=%v err=%v", sess.OrchestratorInfo.Transcoder, err)
```

Comment:
I'm on board with annotating errors. But, as-is this error is not being used/logged anywhere. If we're going to do this I'd prefer if we return annotated errors so that they can be handled by the caller (i.e. for logging) and *not* also log the error in the callee (as is done on L377). In this case, we could log the error in [processSegment()](https://github.com/livepeer/go-livepeer/blob/d3f536acf007f8a98edf76f007f2e6ea5517987d/server/broadcast.go#L301).

Can we do this outside of this PR?

Side note: When we do this, `err` is nil here so it would probably be more informative if the error included indicated that the response was empty like the error that is returned in the current code prior to this PR.

Reply:
Creating an issue to track this https://github.com/livepeer/go-livepeer/issues/1410

Reply:
Thanks for creating the issue.

Looks like this error is annotated, but not the one on L382. Can we either annotate both or leave this error annotation out of this PR?

Reply:
removed annotation for errors , we can display a more informative logging message in processSegment as you describe.